library/src/main/java/com/bumptech/glide/RegistryFactory.java

```diff
@@ -279,11 +279,15 @@ Uri.class, Bitmap.class, new ResourceBitmapDecoder(resourceDrawableDecoder, bitm
           DirectResourceLoader.inputStreamFactory(context);
       ModelLoaderFactory<Integer, AssetFileDescriptor> assetFileDescriptorFactory =
           DirectResourceLoader.assetFileDescriptorFactory(context);
+      ModelLoaderFactory<Integer, Drawable> drawableFactory =
+          DirectResourceLoader.drawableFactory(context);
       registry
           .append(int.class, InputStream.class, inputStreamFactory)
           .append(Integer.class, InputStream.class, inputStreamFactory)
           .append(int.class, AssetFileDescriptor.class, assetFileDescriptorFactory)
           .append(Integer.class, AssetFileDescriptor.class, assetFileDescriptorFactory)
+          .append(int.class, Drawable.class, drawableFactory)
+          .append(Integer.class, Drawable.class, drawableFactory)
           .append(Uri.class, InputStream.class, ResourceUriLoader.newStreamFactory(context))
```

Comment:
**Scalability Issue**: The addition of a new drawable factory directly in the RegistryFactory without considering a pluggable or extensible approach for handling various resource types might limit the scalability of the system. As new resource types or special handling cases are introduced, this class may become bloated and harder to maintain, violating the open/closed principle. <br> **Fix**: Consider implementing a more flexible system for registering and handling different types of resources. This could involve creating a registry or manager class that allows for the dynamic addition of resource handlers or factories. This way, as new resource types are needed, they can be added without modifying the core classes, thus improving the scalability and maintainability of the system. <br> **Code Suggestion**: 
 ```
 +      ModelLoaderFactory<Integer, Drawable> drawableFactory =
 +          DirectResourceLoader.drawableFactory(context);
        registry
            .append(int.class, InputStream.class, inputStreamFactory)
            .append(Integer.class, InputStream.class, inputStreamFactory)
            .append(int.class, AssetFileDescriptor.class, assetFileDescriptorFactory)
            .append(Integer.class, AssetFileDescriptor.class, assetFileDescriptorFactory)
 +          .append(int.class, Drawable.class, drawableFactory)
 +          .append(Integer.class, Drawable.class, drawableFactory)
            .append(Uri.class, InputStream.class, ResourceUriLoader.newStreamFactory(context))
            .append(
                Uri.class,
 ```